modal_proto/api.proto

```diff
@@ -2879,6 +2879,71 @@ message VolumeMount {
   bool read_only = 4;
 }
 
+message VolumePutFiles2Request {
+  // List of files, sorted lexicographically by `path`.
+  repeated File files = 1;
+
+  // The last time the client called `VolumePutFiles2` for this file, it was
+  // told that some blocks were missing. This field contains information
+  // about the client having uploaded those missing blocks.
+  repeated NewBlock new_blocks = 2;
+
+  // If set to true, prevent overwriting existing files. (Note that we don't
+  // allow overwriting existing directories with uploaded files regardless.)
+  bool disallow_overwrite_existing_files = 3;
+
+  message File {
+    // Destination path of the file to be uploaded, including any parent dirs
+    // etc.; for example "foo/bar/baz.txt"
+    string path = 1;
+
+    // The total size of the file, in bytes.
+    uint64 size = 2;
+
+    // SHA-256 checksum of each 8MiB block of the file's contents, in binary
+    // (ie 32 raw bytes) format for compactness.
+    repeated bytes blocks_sha256 = 3;
```

Comment:
Could this be a `repeated Block = 3` instead where 
Block
```
message Block {
    block_sha256 = 1;
    optional bytes put_response = 2; // only set for new blocks when requested
}
```
That would get rid of all the file/block index stuff if I'm not mistaken?